### PR TITLE
jit: enable conv_relu fusion

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Now, only enable conv_relu fusion for jit path.

### CODE REVIEW

**Comment by XiaobingSuper** [comment]
@EikanWang @hongzhen1 @CaoZhongZ

File: torch_ipex/csrc/jit/register_dnnl_jit_ops.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ RegisterOperators op({
         return [] (Stack& stack) {
           at::Tensor input;
           pop(stack, input);
-          auto result = dnnl_relu_(input);
+          auto result = AtenIpexCPUDev::dil_relu_(input);
```
Comment (EikanWang) [inline torch_ipex/csrc/jit/register_dnnl_jit_ops.cpp:63]: What's the behavior if the input tensor is non-contiguous?

File: torch_ipex/csrc/jit/fusion_pass.cpp [hunk 2]
```diff
@@ -198,7 +199,8 @@ class OpFuser {
     }
 
     // throw
-    auto er = script::ErrorReport(node->sourceRange());
+    //auto er = script::ErrorReport(node->sourceRange());
```
Comment (EikanWang) [inline torch_ipex/csrc/jit/fusion_pass.cpp:202]: If the code is invalid, could we remove it?

File: torch_ipex/csrc/jit/fusion_pass.cpp [hunk 3]
```diff
@@ -323,7 +325,7 @@ class OpFuser {
     }
 
     return std::make_pair(++pos->iterator(), changed);
-}
+  }
```
Comment (EikanWang) [inline torch_ipex/csrc/jit/fusion_pass.cpp:328]: seems like these is a redauant whitespace character here?

**Comment by EikanWang** [comment]
If the code is invalid, could we remove it directly?

File: torch_ipex/csrc/jit/fusion_pass.cpp (hunk 3 above)
Comment (XiaobingSuper) [reply torch_ipex/csrc/jit/fusion_pass.cpp:272]: it is not a redauant whitespace.

File: torch_ipex/csrc/jit/register_dnnl_jit_ops.cpp (hunk 1 above)
Comment (XiaobingSuper) [reply torch_ipex/csrc/jit/register_dnnl_jit_ops.cpp]: Solved.

File: torch_ipex/csrc/jit/fusion_pass.cpp (hunk 2 above)
Comment (XiaobingSuper) [reply torch_ipex/csrc/jit/fusion_pass.cpp]: done.

File: torch_ipex/csrc/init_python_bindings.cpp [hunk 4]
```diff
@@ -89,7 +95,7 @@ void InitIpexModuleBindings(py::module m) {
           return AtenIpexTypeExt::embedding_bag_backward(grad_out, weights, inputs, offsets);
         });
   m.def("linear",
-        [](const at::Tensor& input, const at::Tensor& weight, const c10::optional<at::Tensor>& bias) {
+        [](const at::Tensor& input, const at::Tensor& weight, const at::Tensor& bias) {
```
Comment (pinzhenx) [inline torch_ipex/csrc/init_python_bindings.cpp:98]: pybind registry here is no use?

File: torch_ipex/csrc/cpu/CustomerOps.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/autograd/custom_function.h>
+#include <torch/csrc/autograd/function.h>
+#include <ATen/Tensor.h>
+#include <torch/script.h>
+#include <c10/util/Optional.h>
+#include "torch_ipex/csrc/utils.h"
+#include "DevOPs.h"
+
+using namespace at;
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/CustomerOps.h:12]: avoid `using namespace` in header file

File: torch_ipex/csrc/cpu/ExtendOPs.cpp [hunk 6]
```diff
@@ -9,6 +9,7 @@
 #include "xsmm/libxsmm_utils.h"
 #include "../utils.h"
 #include "DevOPs.h"
+#include "CustomerOps.h"
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/ExtendOPs.cpp:12]: minor, `CustomOps`

File: torch_ipex/csrc/cpu/RegisterOps.cpp [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#include <torch/script.h>
+#include "ExtendOPs.h"
+
+static auto registry =
+    torch::RegisterOperators()
+       .op("torch_ipex::linear",
+          [](const at::Tensor& input, const at::Tensor& weight, const at::Tensor& bias) {
+          return torch_ipex::AtenIpexTypeExt::linear(input, weight, bias);
```
Comment (pinzhenx) [inline torch_ipex/csrc/cpu/RegisterOps.cpp:8]: lambda seems to be redundant here

File: torch_ipex/csrc/init_python_bindings.cpp (hunk 4 above)
Comment (XiaobingSuper) [reply torch_ipex/csrc/init_python_bindings.cpp:97]: Just leave it now.

**Comment by XiaobingSuper** [comment]
@EikanWang , please help merge it, thanks!

File: intel_pytorch_extension_py/ops/linear.py [hunk 8]
```diff
@@ -2,30 +2,11 @@
...
-        return (grad_input, grad_weight, grad_bias)
-
-def linear(input, weight, bias=None):
-    if input.device.type == 'dpcpp' and core.get_auto_dnnl():
-        return LinearFunction.apply(input, weight, bias)
-    return F_linear(input, weight, bias)
+def linear(input, weight, bias: Optional[torch.Tensor] = None):
+    if bias is None:
+        bias = torch.zeros(weight.size(0))
+    return torch.ops.torch_ipex.linear(input, weight, bias)
```
Comment (pinzhenx) [inline intel_pytorch_extension_py/ops/linear.py:10]: @XiaobingSuper there's no way to fallback to aten linear if there's any exception thrown our custom linear. I suggest we should add try-catch here? or in the `NewLinearOp`?

File: intel_pytorch_extension_py/ops/pooling.py [hunk 9]
```diff
@@ -65,6 +39,11 @@ def max_pool3d(input, kernel_size, stride, padding, dilation, ceil_mode):
         pass
     return torch_max_pool3d(input, kernel_size, stride, padding, dilation, ceil_mode)
 
+def max_pool2d(input, kernel_size: Vector, stride: Vector, padding: Vector, dilation: Vector, ceil_mode: bool):
```
Comment (pinzhenx) [inline intel_pytorch_extension_py/ops/pooling.py:42]: ditto

File: intel_pytorch_extension_py/ops/pooling.py [hunk 10]
```diff
@@ -41,21 +28,8 @@ def backward(ctx, grad_output):
...
-
-def max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode):
-    try:
-        if input.device.type == 'dpcpp' and core.get_auto_dnnl():
-            return MaxPoolingFunction.apply(input, kernel_size, stride, padding, dilation, ceil_mode)
-    except RuntimeError:
-        pass
-    return torch_max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
+def adaptive_avg_pool2d(input, output_size: Vector):
+    return torch.ops.torch_ipex.adaptive_avg_pool2d(input, _pair(output_size))
```
Comment (pinzhenx) [inline intel_pytorch_extension_py/ops/pooling.py:32]: ditto

**Pull request merged**